src/compiler/transformers/decorators-to-static/convert-decorators.ts

```diff
@@ -143,3 +192,234 @@ const filterDecorators = (
+ *
+ * ```ts
+ * class CompiledMyClass {
+ *   foo = "bar"
+ * }
+ * ```
+ *
+ * This output is more correct because the compiled code 1) more closely
+ * resembles the TypeScript source and 2) is using standard JS syntax instead
+ * of desugaring it. There is an announcement in the release notes for
+ * TypeScript v3.7 which explains some helpful background about the change,
+ * and about the `useDefineForClassFields` TypeScript option which lets you
+ * opt-in to the old output:
+ *
+ * https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#the-usedefineforclassfields-flag-and-the-declare-property-modifier
+ *
+ * For our use-case, however, the ES2022+ behavior doesn't work, since we need
+ * to be able to define getters and setters on these fields. We could require
+ * that the TypeScript configuration used for Stencil have the
+ * `useDefineForClassFields` setting set to `false`, but that would have the
+ * undesirable side-effect that class fields which are _not_
+ * decorated with a Stencil decorator would also be instantiated in the
+ * constructor.
+ *
+ * So instead, we take matters into our own hands. When we encounter a class
+ * field which is decorated with a Stencil decorator we remove it from the
+ * class and add a statement to the constructor to instantiate it with the
+ * correct default value.
+ *
+ * **Note**: this function will modify a constructor if one is already present on
+ * the class or define a new one otherwise.
+ *
+ * @param classNode a TypeScript AST node for a Stencil component class
+ * @param classMembers the class members that we need to update
+ * @returns a list of updated class elements which can be inserted into the class
+ */
+function handleClassFields(classNode: ts.ClassDeclaration, classMembers: ts.ClassElement[]): ts.ClassElement[] {
+  const statements: ts.ExpressionStatement[] = [];
+  const updatedClassMembers: ts.ClassElement[] = [];
+
+  for (const member of classMembers) {
+    if (shouldInitializeInConstructor(member) && ts.isPropertyDeclaration(member)) {
+      const declarationName: ts.DeclarationName = ts.getNameOfDeclaration(member);
+
+      const memberName =
+        declarationName.kind === ts.SyntaxKind.ComputedPropertyName
+          ? declarationName.expression.getText()
+          : declarationName.getText();
+
+      // this is a class field that we'll need to handle, so lets push a statement for
+      // initializing the value onto our statements list
+      statements.push(
+        ts.factory.createExpressionStatement(
+          ts.factory.createBinaryExpression(
+            ts.factory.createPropertyAccessExpression(ts.factory.createThis(), ts.factory.createIdentifier(memberName)),
+            ts.factory.createToken(ts.SyntaxKind.EqualsToken),
+            // if the member has no initializer we should default to setting it to
+            // just 'undefined'
+            member.initializer ?? ts.factory.createIdentifier('undefined')
```

Comment:
hum I don't think so, according to my editor `member.initializer` will be `ts.Expression` (i.e. an AST node representing an expression) or `undefined`, so I'm basically ensure that if it _is_ undefined we hoist that value into AST-land instead of JS-runtime-value-land, if that makes sense? 🤔

Reply:
another way to say that is that yeah, I think if the expression evaluates to `null` that won't come into play until later on in the backend of the TS compiler when it goes to emit JS and has to turn such AST nodes into concrete values in the emitted code